src/plugins/explore/public/application/utils/state_management/actions/set_dataset/set_dataset.ts

```diff
@@ -58,6 +60,14 @@ export const setDatasetActionCreator = (
     dispatch(setPromptModeIsAvailable(newPromptModeIsAvailable));
   }
 
+  const newSummaryAgentIsAvailable = await getSummaryAgentIsAvailable(
```

Comment:
I suggest to do `Promise.allSettled([await getPromptModeIsAvailable(), await getSummaryAgentIsAvailable()])` so the two requests be fired in parallel

Reply:
+1 to @ruanyl , we don't want these two async calls to happen synchronously